protocol/x/vault/keeper/shares.go

```diff
@@ -2,11 +2,13 @@ package keeper
 
```

Comment:
Is it possible to use the E2E testing framework to add an E2E test of the flow of:
- lock shares til block X
- advance to block X
- check that shares are unlocked

Reply:
yeah this will be possible once there's a tx that calls `LockShares`. I was thinking once PML is integrated, we can add this e2e test. (before we have a tx that calls `LockShares`, e2e won't work)

Reply:
added a task [here](https://linear.app/dydx/issue/TRA-585/add-e2e-test-on-shares-locking-once-pml-is-integrated)

Reply:
Are you sure there's a need for a tx that calls `LockShares`? Isn't there an instance of the `VaultsKeeper` that's accessible on the test app in the E2E test framework?

Reply:
we can call `VaultsKeeper.LockShares` keeper method but i don't think the state changes from that call persists once you advance to next block (as it's not part of a tx)

Reply:
Good point, possibly an E2E test with a delayed message w/ unlock shares could be useful (just to check authorities/everything else works) but not necessary.